waves-integration/src/main/scala/com/wavesplatform/dex/domain/bytes/ByteStr.scala

```diff
@@ -39,8 +39,16 @@ final case class ByteStr(arr: Array[Byte]) {
   def takeRight(n: Long): ByteStr = drop(arr.length.toLong - n)
   def dropRight(n: Long): ByteStr = take(arr.length.toLong - n.max(0))
 
+  def canEqual(other: Any): Boolean = other.isInstanceOf[ByteStr]
+
   override def equals(a: Any): Boolean = a match {
-    case other: ByteStr => java.util.Arrays.equals(arr, other.arr)
+    case other: ByteStr =>
+      other.canEqual(this) &&
+        java.util.Arrays.equals(arr, other.arr) &&
+        this.base58.equals(other.base58) &&
+        this.base64Raw.equals(other.base64Raw) &&
+        this.base64.equals(other.base64) &&
+        this.trim.equals(other.trim)
```

Comment:
We don't need this, because these strings are just representation of `arr` bytes

Reply:
fixed

Reply:
Actually, this is not fixed